src/components/ContactInfo/ContactInfo.vue

```diff
@@ -0,0 +1,365 @@
+              >
+                <span>Cancel</span>
+              </v-btn>
+            </div>
+          </v-flex>
+        </v-layout>
+      </v-form>
+    </template>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Prop, Watch, Emit, Vue } from 'vue-property-decorator'
+import { mask } from 'vue-the-mask'
+
+// Interfaces & enums
+import { ContactIF, FormIF } from '@/interfaces'
+
+@Component({
+  directives: { mask }
+})
+export default class ContactInfo extends Vue {
+  // Refs
+  $refs!: Vue['$refs'] & {
+    editContactForm: FormIF,
+  }
+
+  /** The current business contact information */
+  @Prop()
+  private businessContact!: ContactIF
+
+  /** The baseline contact information */
+  @Prop()
+  private originalBusinessContact!: ContactIF
+
+  /** Flag for identifying changes */
+  @Prop()
+  private hasBusinessContactInfoChange!: boolean
+
+  /** Edit label name (ie 'Change' or 'Correct') */
+  @Prop()
+  private editLabel!: string
+
+  /** Edited label name (ie 'Changed' or 'Corrected') */
+  @Prop()
+  private editedLabel!: string
+
+  // Local Properties
+  private isEditing: boolean = false
+  private contactInfo: ContactIF
+  private formValid: boolean = false
+
+  // Text-field Rules
+  private emailRules = []
+  private confirmEmailRules = []
+  private phoneRules = []
+
+  /** V-model for dropdown menu. */
+  private dropdown: boolean = null
+
+  /** Emit the updated data to parent component */
+  private updateContactInfo (): void {
+    console.log('Inside Update')
+    this.emitContactInfo(this.contactInfo)
+    this.isEditing = false
+  }
+
+  /** Cancel the current local edits */
+  private cancelEdit (): void {
+    this.contactInfo = { ...this.businessContact }
+    this.emitContactInfo(this.contactInfo)
+    this.isEditing = false
+  }
+
+  /** Restore the contact info to original values */
+  private resetContactInfo (): void {
+    this.contactInfo = { ...this.originalBusinessContact }
+    this.emitContactInfo(this.contactInfo)
+    this.isEditing = false
+  }
+
+  /** Submit method to handle validations and data update. */
+  private async submitContact (): Promise<void> {
+    // Set Rules on submission
+    this.emailRules = [
+      (v: string) => !!v || 'Email address is required',
+      (v: string) => {
+        const pattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+        return pattern.test(v) || 'Valid email is required'
+      }
+    ]
+    this.confirmEmailRules = [
+      (v: string) => !!v || 'Confirm email address',
+      (v: string) => v.toString() === this.contactInfo.email || 'Email addresses must match'
+    ]
+    this.phoneRules = [
+      (v: any) => !v || (v.length === 0 || v.length === 14) || 'Phone number is invalid'
+    ]
+
+    // Await the applied rules and validate form
+    await Vue.nextTick()
+    this.$refs.editContactForm.validate()
```

Comment:
This is an interesting way to validate only after the user clicks Save. Does it work well?

Have you ever gotten lazy validation to work? In theory that's an alternative.

Reply:
This works pretty well, yea! 
By handling the validations and applying the rules on submit it prevents those validations triggers while a user is typing/editing and so on. 
Very often the feedback from UX is related to way Vue validator `rules` work, so this is just another approach. 
Name request also has another approach but it seemed very verbose, so just tried to keep it simple here and only apply the rules once a user was done with the form. 
Is lazy validation a feature of Vue or is it a alternative package perhaps?

Reply:
https://vuejs.org/v2/guide/forms.html#lazy